app/webpacker/components/Panel/SeniorDelegate/Regions/index.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect, useMemo } from 'react';
+
+import { Dropdown, Header } from 'semantic-ui-react';
+
+import useInputState from '../../../../lib/hooks/useInputState';
+import useLoadedData from '../../../../lib/hooks/useLoadedData';
+import { apiV0Urls } from '../../../../lib/requests/routes.js.erb';
+import { delegateRegionsStatus, groupTypes } from '../../../../lib/wca-data.js.erb';
+import Errored from '../../../Requests/Errored';
+import Loading from '../../../Requests/Loading';
+import Region from './Region';
+
+export default function Regions({ loggedInUserId }) {
+  const {
+    data: seniorDelegateRoles,
+    loading, error,
+  } = useLoadedData(apiV0Urls.userRoles.listOfUser(
+    loggedInUserId,
+    'groupName', // Sort params
+    {
+      isActive: true,
+      isGroupHidden: false,
+      status: delegateRegionsStatus.senior_delegate,
+      groupType: groupTypes.delegate_region,
+    },
+  ));
+  const [selectedGroupIndex, setSelectedGroupIndex] = useInputState();
+
+  useEffect(() => {
+    if (seniorDelegateRoles?.length > 0) setSelectedGroupIndex(0);
+  }, [seniorDelegateRoles, setSelectedGroupIndex]);
+
+  const groupOptions = useMemo(() => seniorDelegateRoles.map((role, index) => ({
```

Comment:
Won't this need
```suggestion
  const groupOptions = useMemo(() => seniorDelegateRoles?.map((role, index) => ({
```
?

Reply:
Yes. To be honest, it was there in my code but uncommitted. Added now.